crates/router/src/core/pm_auth.rs

```diff
@@ -716,6 +717,26 @@ pub async fn retrieve_payment_method_from_auth_service(
         .ok_or(ApiErrorResponse::InternalServerError)
         .attach_printable("Bank account details not found")?;
 
+    if let Some(balance) = bank_account.balance {
+        let required_conversion = util_types::FloatMajorUnitForConnector;
+        let converted_amount = required_conversion
+            .convert_back(
+                balance,
+                payment_intent.currency.unwrap_or(enums::Currency::USD),
+            )
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Could not convert to MinorUnit")?;
```

Comment:
could u change this error message to convey more on what conversion failed. Something like

```suggestion
            .attach_printable("Could not convert {something} to MinorUnit")?;
```